interfaces/builtin/password.go

```diff
@@ -0,0 +1,42 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const passwordConnectedPlugAppArmor = `
+/usr/bin/passwd ix,
+/usr/sbin/adduser ix,
+/usr/sbin/useradd ix,
+/usr/sbin/chpasswd ix,
```

Comment:
Please rewrite these to be more path-agnostic (and include the additional commands I list):

```
# Allow setting the password for users and groups and modifying
# passwd and shadow entries
/{,usr/}bin/passwd ixr,
/{,usr/}bin/gpasswd ixr,
/{,usr/}bin/chage ixr,
/{,usr/}bin/chfn ixr,
/{,usr/}bin/chsh ixr,
/{,usr/}bin/expiry ixr,

# Also allow the batch-mode variants
/{,usr/}sbin/chpasswd ixr,
/{,usr/}sbin/chgpasswd ixr,
/{,usr/}sbin/newusers ixr,

# Allow checking integrity of files
/{,usr/}sbin/pwck ixr,
/{,usr/}sbin/grpck ixr,

# Allow adding, removing and modifying users and groups
/{,usr/}sbin/{add,del}user ixr,
/{,usr/}sbin/user{add,mod,del} ixr,
/{,usr/}sbin/{add,del}group ixr,
/{,usr/}sbin/group{add,mod,del} ixr,
```

Let's leave out the shadowconfig, cppw, cpgr, vipw, vigr and *conv executables for now. I didn't test the above. Please do so and update the apparmor policy accordingly. It's possible some of these commands are not aware of nss-extrausers and if not, please file bugs against the 'shadow' package (and ping ogra).